### DISCUSSION TITLE
Sorting by publication date

### DISCUSSION THREAD

**Discussion by giampieropace** [FreshRSS/FreshRSS#5865] [Q&A] [answered]
I installed today freshRSS and added my feeds, but I'm facing this problem: the feeds are first grouped and then sorted by date.

E.g.:

> feed 1 - title - 11 nov 2023
> feed 1 - title - 10 nov 2023
> feed 1 - title - 09 nov 2023
> feed 1 - title - 08 nov 2023
> feed 2 - title - 11 nov 2023
> feed 2 - title - 10 nov 2023
> feed 2 - title - 09 nov 2023
> feed 2 - title - 08 nov 2023
> and so on...

What I want is to have all in cronological order:

> feed 1 - title - 11 nov 2023
> feed 2 - title - 11 nov 2023
> feed 1 - title - 10 nov 2023
> feed 2 - title - 10 nov 2023
> feed 1 - title - 09 nov 2023
> feed 2 - title - 09 nov 2023
> feed 1 - title - 08 nov 2023
> feed 2 - title - 08 nov 2023

If I click the "older/newest first" button the situation doesn't change, feeds are inverted, but always grouped.

[Here is a short video recording of the issue](https://screenrec.com/share/HGYvbo5hys).

Am I missing something?

**Comment by Alkarex** [comment]
Edit: PR for custom order-by https://github.com/FreshRSS/FreshRSS/pull/7149

Hello,
This behaviour occurs when adding new feeds one at a time (importing several feeds by OPML mitigates the issue).
This is by design at the moment, and the apparent problem becomes reduced when feeds gets regularly refreshed.

The order you see **is** chronological, just the chronology of when articles have been discovered/received (similar to what e-mail clients are doing), not the date the articles *advertise* (the advertised date is not a robust sorting criteria as dates might very well be a long time in the past or in the future, as in common in the case of [academic articles](https://github.com/FreshRSS/FreshRSS/issues/2154#issuecomment-442397876), book publications, events, [chapters of science fiction](https://github.com/FreshRSS/FreshRSS/discussions/5569), [etc](https://github.com/FreshRSS/FreshRSS/issues/4057).)

A longer explanation of the rationale for the current behaviour can be see on e.g.

- https://github.com/FreshRSS/FreshRSS/issues/1771#issuecomment-359195655
- https://github.com/FreshRSS/FreshRSS/issues/2083#issuecomment-433622668

And there is a plan to customise sorting (on different criteria including advertised date):

- https://github.com/FreshRSS/FreshRSS/issues/2596#issuecomment-1541934281

**Reply by MathieuGodard** [reply]
> the advertised date is not a robust sorting criteria as dates might very well be a long time in the past or in the future, as in common in the case of https://github.com/FreshRSS/FreshRSS/issues/2154#issuecomment-442397876, book publications, events, https://github.com/FreshRSS/FreshRSS/discussions/5569, https://github.com/FreshRSS/FreshRSS/issues/4057.

Hello,
Ok, but can we leave the choice to the user by warning them of this behavior? Personally, I only use media feeds, and this feature would be very useful to me!

**Reply by Frenzie** [reply]
That is a (longer term) goal, yes.

**Reply by Alkarex** [reply]
https://github.com/FreshRSS/FreshRSS/pull/7149

**Comment by h4rp00n33r** [comment]
Hey @Alkarex , is it possible to set the order by publication date as default for the web frontend of FreshRSS?

**Comment by shadowlocked** [comment]
I too need sorting by publication date. The only solution I have found is this: https://blog.kekling.com/?p=502

<img width="250" height="119" alt="image" src="https://github.com/user-attachments/assets/c4ab0a37-059d-445e-b8ca-31666b3d35f6" />

I use IIS rather than Apache, but the rules can be imported. But before I reinvent the wheel again (!) is there some way that is not obvious so that the sort order can be by publication date?

**Reply by Alkarex** [reply, marked as answer]
Copy of my response: https://github.com/FreshRSS/FreshRSS/pull/7149#issuecomment-3436189521

> Is there any existing way to address the default setting in config? 

@shadowlocked Yes there is, and more than one way. Since [this PR](https://github.com/FreshRSS/FreshRSS/pull/7149), by changing the corresponding preference in your config file. Alternatively, thanks to the sorting criteria being kept while you navigate between articles / feeds / categories, you can just change once to your favourite sorting criteria. This means that you can even bookmark your FreshRSS by including the sorting criteria in the URL.

See:
* https://github.com/FreshRSS/FreshRSS/issues/8085#issuecomment-3392435306
* https://github.com/FreshRSS/FreshRSS/issues/4761

For the first approach, change the sort order to `date` in your `data/users/shadowlocked/config.php` on the model of:

https://github.com/FreshRSS/FreshRSS/blob/ec1f5ee61bfeb9e8ed6a3c1e069b82d9f26f64e6/config-user.default.php#L56

> For journalists and researchers, anything other than sorting by publication date is not a good default.

*Publication date* is highly misunderstood by many users, which is also a reason why it has not been promoted more. Articles may have publication dates far in the past or in the future (or no date at all), either for legitimate reasons or due to source bugs. A consequence is that the sorting view is not stable when using *publication date*, since new articles will often be inserted in the middle of existing articles, which is quite confusing. This makes it a poor and error-prone default.
The only stable and reliable criteria for sorting by freshness is the received date, which very often will be quite similar to the publication date.

But with that been said, yes, the default sorting criteria can nevertheless already be changed, and there are issues open to contributors for additional UI and options.

**Reply by shadowlocked** [reply]
> _Publication date_ is highly misunderstood by many users, which is also a reason why it has not been promoted more. Articles may have publication dates far in the past or in the future (or no date at all), either for legitimate reasons or due to source bugs.

I guess we can agree to differ on this, though I do note that some others have really wanted this functionality. In any case, by changing line 56 to 	`'sort' => 'date',` in C:\inetpub\wwwroot\FreshRSS\config-user.default.php, I am so far unable to get FreshRSS to not default back to 'Freshly received first'. Since the Apache/IIS rewrite rules _are_ working in this respect, I'll have to leave it there for the moment due to time considerations.

**Reply by Alkarex** [reply]
Please double-check https://github.com/FreshRSS/FreshRSS/discussions/5865#discussioncomment-14761591

> For the first approach, check the sort order in your `data/users/shadowlocked/config.php` on the model of:
> 
> https://github.com/FreshRSS/FreshRSS/blob/ec1f5ee61bfeb9e8ed6a3c1e069b82d9f26f64e6/config-user.default.php#L56

is not the same than:

> by changing line 56 to `'sort' => 'date',` in C:\inetpub\wwwroot\FreshRSS\config-user.default.php

This is not the correct file.
I have just double-checked, and it works correctly if you change the user's config file, **not** that default file.

**Reply by shadowlocked** [reply]
> I have just double-checked, and it works correctly if you change the user's config file, **not** that default file.

Thank you. I was basing that off your [post ](https://github.com/FreshRSS/FreshRSS/issues/8085#issuecomment-3392435306)of two weeks ago, but now that I have changed the correct file that you have pointed out, and turned off the IIS rewrites, it appears to work. It would be great to get this as a proper control in GUI prefs, since this thread (and those it links to) are probably currently the best documentation for this feature.

**Comment by DDzwiedziu** [comment]
Sorry for thread necromancy, yet can you set the publication date as the default? Settings -> Configuration -> Reading -> View -> Sorting only allows for newest/oldest first.

**Reply by Alkarex** [reply]
Please check higher up: https://github.com/FreshRSS/FreshRSS/discussions/5865#discussioncomment-14761591

By the way, for users with a MySQL database (which is our least favoured database), there was a bug, which is fixed in https://github.com/FreshRSS/FreshRSS/pull/8223

**Comment by Alkarex** [comment]
Work in progress: https://github.com/FreshRSS/FreshRSS/pull/8234
Tests welcome
